lib/ohai/plugins/joyent.rb

```diff
@@ -25,24 +25,11 @@
   depends 'os', 'platform', 'virtualization'
 
   def collect_product_file
-    lines = []
-    if ::File.exists?("/etc/product")
-      ::File.open("/etc/product") do |file|
-        while line = file.gets
-          lines << line
-        end
-      end
-    end
-    lines
+    ::File.read("/etc/product").strip.split("\n") rescue []
   end
 
   def collect_pkgsrc
-    if File.exist?('/opt/local/etc/pkg_install.conf')
-      sm_pkgsrc = ::File.read("/opt/local/etc/pkg_install.conf").split("=")
-      sm_pkgsrc[1].chomp
-    else
-      nil
-    end
+    ::File.read("/opt/local/etc/pkg_install.conf").strip.split("=", 2).last rescue nil
```

Comment:
Ohai catches all exceptions at a higher layer, which I dislike but we can't change that now. So any uncaught exception in a plugin will be rescued and cause a debug message to be printed, so if this plugin were to fail because, say, we had unexpected content in `/opt/local/etc/pkg_install.conf`, at least you would be able to get some feedback about why you suddenly didn't get the ohai data you expected. Which illustrates why I prefer not to do "blind" rescues like this (except in special cases), it hides or obfuscates the true source of the problem when something goes wrong.